### PR TITLE
Refactored out filtering

### PR DESCRIPTION
**Pull request by tisnik**
# Description

Refactored out filtering

Basically the most important functions in `differ.go` code use global module variables so it's quite hard to cover them correctly by using unit tests. It would be nice to refactore such long and complicated code. This is the first attempt, just for the filter. The refactored function is isolated from global module variables and thus testable.

## Type of change

- Refactor (refactoring code, removing useless files)
- Unit tests (no changes in the code)

## Testing steps

Done on CI

## Checklist
* [ ] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [ ] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: differ/differ.go [hunk 1]
```diff
@@ -244,6 +244,26 @@ func findRuleByNameAndErrorKey(
 	return
 }
 
+// evaluateFilterExpression function tries to evaluate event filter expression
+// based on provided threshold values and actual reccomendation values
+func evaluateFilterExpression(eventFilter string,
+	likelihoodThreshold int, impactThreshold int, severityThreshold int, totalRiskThreshold int,
```
Comment (juandspy) [inline differ/differ.go:250]: all these "int" can be merged into just one at the end of the parenthesis

File: differ/differ.go [hunk 2]
```diff
@@ -244,6 +244,26 @@ func findRuleByNameAndErrorKey(
 	return
 }
 
+// evaluateFilterExpression function tries to evaluate event filter expression
+// based on provided threshold values and actual reccomendation values
+func evaluateFilterExpression(eventFilter string,
```
Comment (juandspy) [inline differ/differ.go:249]: I'd use a struct instead of so many parameters

File: differ/differ.go (hunk 1 above)
Comment (tisnik) [reply differ/differ.go]: fixed

File: differ/differ.go (hunk 2 above)
Comment (tisnik) [reply differ/differ.go]: fixed

**Review by juandspy — APPROVED** [review]

**Review by dpensi — APPROVED** [review]

**Review by Bee-lee — APPROVED** [review]
looks OK

**Pull request merged**